app/src/main/java/com/gmail/sanovikov71/githubclient/data/DataService.java

```diff
@@ -20,83 +24,136 @@
-//        if (ACTION_FETCH.equals(intent.getAction())) {
-//            fetchUsers();
-//        }
-//
-//        return START_NOT_STICKY;
-//    }
-
-    public void fetchUsers(final UiElement ui) {
+    public void fetchUsers(final ProgressBarUiElement ui) {
         fetch(ui, 0);
     }
 
-    public void fetchMoreUsers(final UiElement ui, int offset) {
-        Log.i("Novikov", "fetchMoreUsers with offset: " + offset);
+    public void fetchMoreUsers(final ProgressBarUiElement ui, int offset) {
         fetch(ui, offset);
     }
 
-    public void fetch(final UiElement ui, int since) {
-        System.out.println("Novikov fetch");
-        Log.i("Novikov", "fetch");
+    public void fetch(final ProgressBarUiElement ui, int since) {
         mGithub.fetchUsers(since).enqueue(new Callback<List<User>>() {
             @Override
             public void onResponse(Call<List<User>> call, Response<List<User>> response) {
-                System.out.println("Novikov onResponse");
                 ui.hideProgressDialog();
 
                 final List<User> body = response.body();
                 if (response.isSuccessful()) {
-                    System.out.println("Novikov good");
+                    final int size = body.size();
+                    ContentValues userList[] = new ContentValues[size];
+                    for (int i = 0; i < size; i++) {
+                        ContentValues userValues = new ContentValues();
+                        final User user = body.get(i);
+                        userValues.put(UserEntry.COLUMN_GITHUB_ID, user.getId());
+                        userValues.put(UserEntry.COLUMN_LOGIN, user.getLogin());
+                        userValues.put(UserEntry.COLUMN_AVATAR_URL, user.getAvatarUrl());
+                        userList[i] = userValues;
+                    }
+
+                    getContentResolver()
+                            .bulkInsert(UserEntry.CONTENT_URI, userList);
+
                 } else {
-                    System.out.println("Novikov bad");
                     ui.showError(response.code());
                 }
-
-                final int size = body.size();
-                ContentValues userList[] = new ContentValues[size];
-                for (int i = 0; i < size; i++) {
-                    ContentValues cardValues = new ContentValues();
-                    final User user = body.get(i);
-                    cardValues.put(UserEntry.COLUMN_LOGIN, user.getLogin());
-                    cardValues.put(UserEntry.COLUMN_AVATAR_URL, user.getAvatarUrl());
-                    userList[i] = cardValues;
-                }
-
-                getContentResolver()
-                        .bulkInsert(UserEntry.CONTENT_URI, userList);
             }
 
             @Override
             public void onFailure(Call<List<User>> call, Throwable t) {
-                System.out.println("Novikov onFailure");
                 ui.hideProgressDialog();
```

Comment:
As you're triggering ui from service this indicates that you're missing some abstractions to not do that.